<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [mod_python] Mod_python Charts
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mod_python%40modpython.org?Subject=%5Bmod_python%5D%20Mod_python%20Charts&In-Reply-To=B3DF88A6-68AB-4AEE-A177-8D2EA2A63AEF%40dscpl.com.au">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="019269.html">
   <LINK REL="Next"  HREF="019264.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[mod_python] Mod_python Charts</H1>
<table border=0 width="100%"><tr><td valign="top">
    <B>Jim Gallacher</B> 
    <A HREF="mailto:mod_python%40modpython.org?Subject=%5Bmod_python%5D%20Mod_python%20Charts&In-Reply-To=B3DF88A6-68AB-4AEE-A177-8D2EA2A63AEF%40dscpl.com.au"
       TITLE="[mod_python] Mod_python Charts">jg.lists at sympatico.ca
       </A><BR>
    <I>Fri Oct 14 18:32:22 EDT 2005</I>
    <P><UL>
        <LI>Previous message: <A HREF="019269.html">[mod_python] Mod_python Charts
</A></li>
        <LI>Next message: <A HREF="019264.html">[mod_python] Mod_python Charts
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#19271">[ date ]</a>
              <a href="thread.html#19271">[ thread ]</a>
              <a href="subject.html#19271">[ subject ]</a>
              <a href="author.html#19271">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Graham Dumpleton wrote:
&gt;<i> 
</I>&gt;<i> On 14/10/2005, at 10:13 PM, &lt;<A HREF="http://mailman.modpython.org/mailman/listinfo/mod_python">python_it at hotmail.com</A>&gt; wrote:
</I>&gt;<i> 
</I>&gt;&gt;<i> Did somebody has experience with CharDirector and Mod_python  
</I>&gt;&gt;<i> (Publisher).
</I>&gt;&gt;<i> (Python 2.4 / ChartDirector for Python / Modpython 3.2.2b)
</I>&gt;&gt;<i> I can&#8217;t make chart with this combination? (I don&#8217;t get errors?)
</I>&gt;&gt;<i> Publisher is working ok with other scripts.
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> I use the following script files:
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> Index.htm =&gt; index.py =&gt; chart.tmpl
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> Index.htm
</I>&gt;&gt;<i> =================
</I>&gt;&gt;<i> Link to index.py
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> Index.py
</I>&gt;&gt;<i> =================
</I>&gt;&gt;<i> from mod_python import psp,util
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> def index(req, perc=[10 , 10 , 80], descrip=['10', '10' ,'80'],  
</I>&gt;&gt;<i> title=&quot;chart&quot;, filename=&quot;chart.png&quot;):
</I>&gt;&gt;<i>     req.content_type = &#8216;text/html&#8217;
</I>&gt;&gt;<i>     tmpl = psp.PSP(req, filename='templates/chart.tmpl', vars= {'req': 
</I>&gt;&gt;<i> req, 'psp': psp, 'perc': perc, 'descrip': descrip, 'title':  title, 
</I>&gt;&gt;<i> 'filename': filename})
</I>&gt;&gt;<i>     tmpl.run()
</I>&gt;<i> 
</I>&gt;<i> 
</I>&gt;<i> For the filename argument you are using a relative pathname. This will
</I>&gt;<i> not work as Apache often runs with cwd as '/' or some other directory.
</I>&gt;<i> It will not be the directory the script file is in and changing cwd  should
</I>&gt;<i> not be done. Instead use:
</I>&gt;<i> 
</I>&gt;<i>   import os
</I>&gt;<i>   __here__ = os.path.dirname(__file__)
</I>&gt;<i> 
</I>&gt;<i>   from mod_python import psp,util
</I>&gt;<i> 
</I>&gt;<i>   def index(req, perc=[10 , 10 , 80], descrip=['10', '10' ,'80'],  
</I>&gt;<i> title=&quot;chart&quot;, filename=&quot;chart.png&quot;):
</I>&gt;<i>       req.content_type = &#8216;text/html&#8217;
</I>&gt;<i>       filename = os.path.join(__here__,'templates/chart.tmpl')
</I>&gt;<i>       tmpl = psp.PSP(req, filename=filename vars={'req': req, 'psp':  
</I>&gt;<i> psp, 'perc': perc,'descrip': descrip, 'title': title, 'filename':  
</I>&gt;<i> filename})
</I>&gt;<i>       tmpl.run()
</I>&gt;<i> 
</I>
Furthermore, ChartDirector does not seem to raise an exception if it 
can't create the file. Testing on linux using the original code, the 
file was not created (no write permission on '/' for the www-data 
process) and no exception was raised. This certainly goes against my 
expectations.

Jim

</PRE>


<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="019269.html">[mod_python] Mod_python Charts
</A></li>
	<LI>Next message: <A HREF="019264.html">[mod_python] Mod_python Charts
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#19271">[ date ]</a>
              <a href="thread.html#19271">[ thread ]</a>
              <a href="subject.html#19271">[ subject ]</a>
              <a href="author.html#19271">[ author ]</a>
         </LI>
       </UL>

</td>
<td align="right" valign="top">
<script type="text/javascript"><!--
google_ad_client = "pub-9718360309690383";
google_ad_width = 120;
google_ad_height = 600;
google_ad_format = "120x600_as";
google_color_border = "CCCCCC";
google_color_bg = "FFFFFF";
google_color_link = "000000";
google_color_url = "666666";
google_color_text = "333333";
//--></script>
<script type="text/javascript"
  src="http://pagead2.googlesyndication.com/pagead/show_ads.js">
</script>
</td>
</table>

<hr>
<a href="http://mailman.modpython.org/mailman/listinfo/mod_python">More information about the Mod_python
mailing list</a><br>
</body></html>
